protobuf/sync/v1/sync_service.proto

```diff
@@ -6,10 +6,18 @@ option go_package = "flagd/grpcsync";
 
 // SyncFlagsRequest is the request initiating the sever-streaming rpc. Flagd sends this request, acting as the client
 message SyncFlagsRequest {
-  // Optional: A unique identifier for flagd provider (grpc client) initiating the request. The server implementations
-  // can utilize this identifier to aggregate flag configurations and stream them to a specific client. This identifier
-  // is intended to be optional. However server implementation may enforce it.
+  // Optional: A unique identifier for flagd(grpc client) initiating the request. The server implementations may
+  // utilize this identifier to uniquely identify, validate(ex:- enforce authentication/authorization) and filter
+  // flag configurations that it can expose to this request. This field is intended to be optional. However server
+  // implementations may enforce it.
+  //    ex:- provider_id: flagd-weatherApp-sidecar
   string provider_id = 1;
+
+  // Optional: A filtering key for the flag configuration request. The server implementation may utilize this to filter
+  // flag configurations from a collection, select the source of the flag or combine this to any desired underlying
+  // filtering mechanism.
+  //    ex:- filtering: 'source=database,key=weatherApp'
+  string filtering = 2;
```

Comment:
How do you feel about `filter`?

Reply:
Yes, that's better. I will update :)

Reply:
Done :)

Reply:
Can you give an example? I could see filter being nearly anything from a key/value pair to a UUID. What would a configuration add?

Reply:
the term filter feels as though other configurations are being excluded, which may not be the case for some implementations, and as such feels mismatched. I do like the suggestion below to use `selector`